### PR TITLE
BHBC 1015 + 1016 - Structure and data passing for creating surveys in the UIs

### PR DESCRIPTION
**Pull request by anissa-agahchen**
# Overview

## Links to jira tickets

- BHBC-1015: Create Survey Base Structure
- BHBC-1016: Create survey section template

## This PR contains the following changes

- structure to house the Surveys List page
- route for the create survey page
- basic container for suvery content

## This PR contains the following types of changes

- [x] New feature (change which adds functionality)

## Checklist

A list of items that are good to consider when making any changes.

_Note: this list is not exhaustive, and not all items are always applicable._

### General

- [x] I have performed a self-review of my own code

### Tests

- [x] I have added/updated unit tests for any code I've added/updated
Note: needs more tests .. but will be in a near future PR

### Linting/Formatting

- [] I have run the linter and fixed any issues, as needed  
       _See the `lint` commands in package.json_
- [x] I have run the formatter and fixed any issues, as needed  
       _See the `format` commands in package.json_

### SonarCloud

- [ ] I have addressed all SonarCloud Bugs, Vulnerabilities, Security Hotspots, and Code Smells

## How Has This Been Tested?

Please describe the tests that you ran to verify your changes.

## Screenshots

Please add any relevant UI screenshots, if applicable.

Screen Shot 2021-05-03 at 10 57 04 AM 4
Screen Shot 2021-05-03 at 11 30 54 AM

### CODE REVIEW

File: app/src/features/projects/view/ProjectSurveysListPage.test.tsx [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+import ProjectSurveysListPage from './ProjectSurveysListPage';
+import { IGetProjectForViewResponse } from 'interfaces/useProjectApi.interface';
+
+const history = createMemoryHistory();
+
+describe('ProjectSurveysListPage', () => {
+  it('renders correctly', () => {
+    const { getByText } = render(
+      <Router history={history}>
+        <ProjectSurveysListPage projectForViewData={{ id: 1 } as IGetProjectForViewResponse} />
```
Comment (sdevalapurkar) [inline app/src/features/projects/view/ProjectSurveysListPage.test.tsx]: we have a sample `IGetProjectForViewResponse` inside `project-helpers.ts`
we should use it here

File: app/src/features/projects/view/ProjectSurveysListPage.tsx [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+const ProjectSurveysListPage: React.FC<IProjectSurveysListPageProps> = (props) => {
+  const history = useHistory();
+  const { projectForViewData } = props;
+
+  const projectId = projectForViewData.id;
+
+  const navigateToCreateSurveyPage = (projectId: number) => {
+    history.push(`/projects/${projectId}/survey/create`);
+  };
+  // TODO: Replace this with the result of an API call giving us back the surveyList data
```
Comment (sdevalapurkar) [inline app/src/features/projects/view/ProjectSurveysListPage.tsx:27]: super nit:
```suggestion
  
  // TODO: Replace this with the result of an API call giving us back the surveyList data
```

File: app/src/features/surveys/CreateSurveyPage.tsx [hunk 3]
```diff
@@ -0,0 +1,182 @@
...
+    }
+  },
+
+  sectionDivider: {
+    height: '1px'
+  }
+}));
+
+/**
+ * Page to display user management data/functionality.
```
Comment (sdevalapurkar) [inline app/src/features/surveys/CreateSurveyPage.tsx]: seems like an incorrect comment

File: app/src/features/surveys/CreateSurveyPage.tsx [hunk 4]
```diff
@@ -0,0 +1,182 @@
...
+ *
+ * @return {*}
+ */
+const CreateSurveyPage: React.FC = () => {
```
Comment (sdevalapurkar) [inline app/src/features/surveys/CreateSurveyPage.tsx]:
```suggestion
const CreateSurveyPage = () => {
```

File: app/src/features/surveys/CreateSurveyPage.tsx [hunk 5]
```diff
@@ -0,0 +1,182 @@
...
+
+    setProjectWithDetails(projectWithDetailsResponse);
+  }, [biohubApi.project, urlParams]);
+
+  useEffect(() => {
+    if (!isLoadingProject && !projectWithDetails) {
+      getProject();
+      setIsLoadingProject(true);
+    }
+  }, [isLoadingProject, projectWithDetails, getProject]);
```
Comment (sdevalapurkar) [inline app/src/features/surveys/CreateSurveyPage.tsx:111]: I dont think we need to do this call

File: app/src/features/surveys/CreateSurveySection.tsx [hunk 6]
```diff
@@ -0,0 +1,66 @@
...
+      }
+
+      setCodes(codesResponse);
+    };
+
+    if (!isLoadingCodes && !codes) {
+      getCodes();
+      setIsLoadingCodes(true);
+    }
+  }, [urlParams, biohubApi.codes, isLoadingCodes, codes]);
```
Comment (sdevalapurkar) [inline app/src/features/surveys/CreateSurveySection.tsx]: we can just pass in codes to this component

**Review by sdevalapurkar — CHANGES_REQUESTED** [review]

File: app/src/features/surveys/SurveysLayout.tsx [hunk 7]
```diff
@@ -0,0 +1,35 @@
+import Box from '@material-ui/core/Box';
```
Comment (NickPhura) [inline app/src/features/surveys/SurveysLayout.tsx]: This file appears to be identical to the ProjectLayout.tsx.  I think for now we can delete this, and just use the ProjectLayout.  

(Even the projectLayout isn't really doing much, but we can address that issue separately later, as needed)

**Review by sdevalapurkar — DISMISSED** [review]
nice

File: app/src/features/surveys/CreateSurveyPage.tsx [hunk 8]
```diff
@@ -0,0 +1,182 @@
...
+
+    '&:last-child': {
+      marginBottom: 0
+    },
+
+    '&:first-child': {
+      marginTop: 0
+    }
+  },
+
```
Comment (sdevalapurkar) [inline app/src/features/surveys/CreateSurveyPage.tsx]:
```suggestion
```

File: app/src/features/surveys/CreateSurveyPage.tsx [hunk 9]
```diff
@@ -0,0 +1,182 @@
...
+    backgroundColor: 'transparent'
+  },
+  surveySection: {
+    marginTop: theme.spacing(4),
+    marginBottom: theme.spacing(5),
+
+    '&:last-child': {
+      marginBottom: 0
+    },
+
```
Comment (sdevalapurkar) [inline app/src/features/surveys/CreateSurveyPage.tsx]:
```suggestion
```

File: app/src/features/surveys/CreateSurveyPage.tsx [hunk 10]
```diff
@@ -0,0 +1,182 @@
...
+  const biohubApi = useBiohubApi();
+
+  const [isLoadingProject, setIsLoadingProject] = useState(false);
+  const [projectWithDetails, setProjectWithDetails] = useState<IGetProjectForViewResponse | null>(null);
+
+  const [isLoadingCodes, setIsLoadingCodes] = useState(false);
+  const [codes, setCodes] = useState<IGetAllCodeSetsResponse>();
+
+  const history = useHistory();
```
Comment (sdevalapurkar) [inline app/src/features/surveys/CreateSurveyPage.tsx:71]:
```suggestion
  const history = useHistory();
  
```

File: app/src/features/surveys/CreateSurveyPage.tsx [hunk 11]
```diff
@@ -0,0 +1,182 @@
...
+  const handleCancel = () => {
+    history.push(`/projects/${projectWithDetails?.id}/surveys`);
+  };
```
Comment (sdevalapurkar) [inline app/src/features/surveys/CreateSurveyPage.tsx:75]:
```suggestion
  };
  
```

**Review by sdevalapurkar — APPROVED** [review]

**Comment by sdevalapurkar** [comment]
duplication is a larger refactoring - can be done later

missing tests will be added in subsequent pr :) @anissa-agahchen

**Pull request closed without merging**